juju/juju
# feat: refactor secret service and secret backend service to use the new atomic run

This PR refactored `secret` and `secretbackend` domain packages to run state calls in the same transaction using the new atomic run.
No feature changes.

## Checklist


## QA steps

unit tests and bash tests.

```
go test -race -v ./domain/secret/... -test.timeout=600s -check.v

go test -race -v ./domain/secretbackend/... -test.timeout=600s -check.v

bash -ex ./main.sh -v -c aws -p ec2 -x output.txt -s '""' secrets_iaas test_secrets_vault

bash -ex ./main.sh -v -c microk8s -p k8s -x output.txt -s '""' secrets_k8s test_user_secrets
```


## Documentation changes

No

## Links

**Jira card:** [JUJU-6268](https://warthogs.atlassian.net/browse/JUJU-6268)



[JUJU-6268]: https://warthogs.atlassian.net/browse/JUJU-6268?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
